docs/DeveloperGuide.md

```diff
@@ -1,34 +1,384 @@
+This feature allows the user to add modules and lesson slots into the timetable.
+Users can also add bookmarks to specific modules and slots.
+Users can enter one-shot-commands, adding multiple slots and bookmarks to a module.
+
+Given below is a sequence diagram of how the feature works.
+![](https://github.com/AY2021S1-CS2113T-W11-1/tp/blob/master/docs/diagrams/addSlotSequenceDiagram.png?raw=true)
+*<center/> Figure 2.3 Sequence diagram for AddSlotCommand </center> <br/></br>*
+
+1. After calling execute() method of the AddSlotCommand object, there will be a check on whether the module code entered by the user already exists in the timetable. If it does not exist, then the module will be created.
+
+2. There will then be a check for additional commands pertaining to the module entered by the user.
+
+3. The code will then check if the command is to add a module bookmark or a lesson slot, and do so accordingly.
+
+4. If the command is to add a lesson slot, then there will be check for a bookmark entry in the command. If there is one, then the bookmark will be added to the lesson slot.
+
+5. Loop to step 3 if there are additional commands which have not been executed.
+
+#### Design consideration:
+
+##### Aspect: How to enable fast typing users to add modules, slots and related bookmarks faster
+* **Alternative 1 (Current choice):** allow one shot command to add slots and bookmarks to a module
+    * Pros: Fast typers can input faster
+    * Cons: Difficult to implement
+* **Alternative 2:** separate adding of modules, lesson slots and related bookmarks into different commands
+    * Pros: Easy to implement
+    * Pros: Lower chance of error
+    * Cons: User has to enter multiple commands each at a time to perform the functions, which takes up more time.
+
+### Edit Slot feature (Francisco)
+
+This feature allows users to edit a slot's title or time. Users can also move slots over to another module.
+
+Given below is an example usage scenario and how the edit mechanism works.
+
+1. The user enters "edit title mon 2 new_title"
+2. A new EditSlotCommand instance is created and the execute() method is called.
+3. The slot corresponding to the day and index given in the user input is retrieved by calling the getSlotByIndexInDay method of the timetable. In this case, the 2nd slot on Monday will be returned.
+4. Based on the chosen field in the user input, different methods are called:
+    a. If the command is "edit module", the moveSlotToAnotherModule method in timetable is called to move the slot to a given module.
+    b. If the command is "edit title", the setTitle method of the retrieved slot is called.
+    c. If the command is "edit time", changeSlotTime is called. This method will call setDay(), setStartTime(), and setEndTime() of the retrieved slot.
+
+The sequence diagram below explains how this feature is executed:
+
+ ![](https://raw.githubusercontent.com/fchensan/tp/docs-images/docs/images/editslotsequence.png)
```

Comment:
Would it be better to standardise with other features the format of whether the diagram or explanation is shown first?